lib/matplotlib/cbook/__init__.py

```diff
@@ -2031,14 +2031,35 @@ def _setattr_cm(obj, **kwargs):
     """
     Temporarily set some attributes; restore original state at context exit.
     """
+    import inspect
     sentinel = object()
-    origs = [(attr, getattr(obj, attr, sentinel)) for attr in kwargs]
+    origs = {}
+    for attr in kwargs:
+        orig = getattr(obj, attr, sentinel)
+        # monkey patching on a new attribute, this is OK
```

Comment:
perhaps you can reorder this to make the common case first, e.g.
```
if (attr in obj.__dict__ or orig is sentinel
        or isinstance(getattr(type(obj), attr), (property, types.FunctionType)):
    origs[attr] = orig
else:
    raise ValueError(...)
```
Checking to types.FunctionType in the type is the same as checking for methods, and (as noted in the other PR, I think) won't give a wrong result if an unrelated bound method is assigned as instance var (`self.foo = otherobj.method`) -- at least I think this point should be fixed.

Reply:
That makes sense. I agree the method from another object is a valid concern (sorry I didn't fully register it before).